src/Calculator/calculator.cpp

```diff
@@ -0,0 +1,140 @@
+#include "include/calculator.h"
+#include <iostream>
+#include <algorithm>
+#include <sstream>
+#include <iterator>
+
+Calculator::Calculator()
+{
+}
+
+double Calculator::calculate(std::string input)
+{
+    clear();
+
+    for (auto token : tokenize(input)) {
+
+        if (isOperand(token)) {
+            handleOperandToken(token);
+            continue;
+        }
+
+        if (isOperator(token)) {
+            if (!handleOperatorToken(token)) {
+                return {};
+            }
+            continue;
+        }
+
+        m_error = "input simbol: " + token + " not supported!";
```

Comment:
suggest to extract token validation to separate loop and use calculation loop without error handling for this.